sql/parser/create.go

```diff
@@ -187,15 +187,21 @@ type IndexTableDef struct {
 
 func (node *IndexTableDef) String() string {
 	var buf bytes.Buffer
-	if node.Name != "" {
-		fmt.Fprintf(&buf, "CONSTRAINT %s ", node.Name)
-	}
 	if node.PrimaryKey {
```

Comment:
this feels ad-hoc:
- parsing CONSTRAINT and INDEX into the same structure is unexpected
- node.PrimaryKey and node.Unique are mutually exclusive, they should be replaced with an enum

I'd really like to see a separation between IndexTableDef and ConstraintTableDef or the removal of IndexTableDef in favour of ADD INDEX.